iodata/formats/mwfn.py

```diff
@@ -0,0 +1,367 @@
+# For pure shells, the order is
+# D shell: D 0, D+1, D-1, D+2, D-2
+# F shell: F 0, F+1, F-1, F+2, F-2, F+3, F-3
+# G shell: G 0, G+1, G-1, G+2, G-2, G+3, G-3, G+4, G-4
+
+CONVENTIONS = {
+    (9, 'p'): HORTON2_CONVENTIONS[(9, 'p')],
+    (8, 'p'): HORTON2_CONVENTIONS[(8, 'p')],
+    (7, 'p'): HORTON2_CONVENTIONS[(7, 'p')],
+    (6, 'p'): HORTON2_CONVENTIONS[(6, 'p')],
+    (5, 'p'): HORTON2_CONVENTIONS[(5, 'p')],
+    (4, 'p'): HORTON2_CONVENTIONS[(4, 'p')],
+    (3, 'p'): HORTON2_CONVENTIONS[(3, 'p')],
+    (2, 'p'): HORTON2_CONVENTIONS[(2, 'p')],
+    (0, 'c'): ['1'],
+    (1, 'c'): ['x', 'y', 'z'],
+    (2, 'c'): ['xx', 'yy', 'zz', 'xy', 'xz', 'yz'],
+    (3, 'c'): ['xxx', 'yyy', 'zzz', 'xyy', 'xxy', 'xxz', 'xzz', 'yzz', 'yyz', 'xyz'],
+    (4, 'c'): HORTON2_CONVENTIONS[(4, 'c')][::-1],
+    (5, 'c'): HORTON2_CONVENTIONS[(5, 'c')][::-1],
+    (6, 'c'): HORTON2_CONVENTIONS[(6, 'c')][::-1],
+    (7, 'c'): HORTON2_CONVENTIONS[(7, 'c')][::-1],
+    (8, 'c'): HORTON2_CONVENTIONS[(8, 'c')][::-1],
+    (9, 'c'): HORTON2_CONVENTIONS[(9, 'c')][::-1],
+}
+
+
+def _load_helper_opener(lit: LineIterator, keys: list) -> Tuple[int, float, float,
+                                                                float, float, float, int]:
+    """Read initial variables."""
+    max_count = len(keys)
+    count = 0
+    d = {}
+    while count < max_count:
+        line = next(lit)
+        for name in keys:
+            if name in line:
+                d[name] = line.split('=')[1].strip()
+                count += 1
+    return int(d['Wfntype']), float(d['Charge']), float(d['Naelec']), float(d['Nbelec']),\
+        float(d['E_tot']), float(d['VT_ratio']), int(d['Ncenter'])
+
+
+def _load_helper_basis(lit: LineIterator) -> Tuple[int, int, int, int, int]:
+    """Read initial variables."""
+    # Nprims must be last or else it gets read in with Nprimshell
+    basis_keywords = ["Nbasis", "Nindbasis", "Nshell", "Nprimshell", "Nprims", ]
+    max_count = len(basis_keywords)
+    count = 0
+    d = {}
+    next(lit)
+    while count < max_count:
+        line = next(lit)
+        for name in basis_keywords:
+            if name in line:
+                d[name] = int(line.split('=')[1].strip())
+                count += 1
+                break
+    return d['Nbasis'], d['Nindbasis'], d['Nprims'], d['Nshell'], d['Nprimshell']
+
+
+def _load_helper_atoms(lit: LineIterator, num_atoms: int) -> Tuple[np.ndarray, np.ndarray,
+                                                                   np.ndarray]:
+    """Read the coordinates of the atoms."""
+    atnums = np.empty(num_atoms, int)
+    atcorenums = np.empty(num_atoms, float)
+    atcoords = np.empty((num_atoms, 3), float)
+    line = next(lit)
+    while '$Centers' not in line and line is not None:
+        line = next(lit)
+
+    for atom in range(num_atoms):
+        line = next(lit)
+        atnums[atom] = int(line.split()[2].strip())
+        atcorenums[atom] = float(line.split()[3].strip())
```

Comment:
Could you use `words = line.spit` upfront and use in all lines below? This is a bit nicer than splitting the same line several times.